### ISSUE TITLE
How to combine multiple loss functions?

### ISSUE THREAD

**Issue by prnvjb** [open-mmlab/mmsegmentation#779]
In the Unet++ paper(https://arxiv.org/pdf/1807.10165.pdf), they stated that
''We have added a combination of binary cross-entropy and dice coefficient as the loss function to each of the above four semantic levels"

However I want to try the same mechanism on other architectures. As I'm new to mmseg; how could I combine multiple loss functions?

thanks for your help!!

**Comment by Junjun2016** [comment]
Hi @prnvjb 
Mixed loss is not supported yet, you can use auxiliary loss or modify the code based on this PR (#244 )!

Related issue #727.

**Comment by MengzhangLI** [comment]
Hi, @prnvjb 
Have you solved this problem?

Right now MMSegmentation does not support multiple loss officially (but it is on the way).

You could just git clone my branch here: https://github.com/open-mmlab/mmsegmentation/pull/818 . My implementation is based on latest code. Thus you shall meet little problems. You are right, combination of Dice + Cross Entropy loss is default setting, and the pull request above is my results. Hoping it is useful to you.

By the way, UNet++ is not a great model because personally I think it just uses more modules (which would add GPU memory) to achieve little numerical results improvements. The better pre and post processing of framework, the less improvements it would get from standard UNet. I said a lot to you because I used its https://github.com/MrGiovanni/UNetPlusPlus/tree/master/pytorch (aka. nnUNet++) but it could not run normally.
